modules/github_integration/frontend/module/git-actions/git-actions.service.ts

```diff
@@ -36,14 +36,12 @@ export class GitActionsService {
     // See https://stackoverflow.com/a/3651867 for how these rules came in.
     // This sanitization tries to be harsher than those rules
     return str
-      .replace(/&/g, 'and ') // & becomes and
-      .replace(/ +/g, '-') // Spaces become dashes
-      .replace(/[\x00-\x1F]/g, '') // ASCII control characters are out
-      .replace(/\x7F/g, '') // DEL is out
-      .replace(/[#\\\/\?\*\~\^\:\{\}@\.\[\]'"]/g, '') // Some other characters with special rules are out
-      .replace(/^[-]+/g, '') // Dashes at the start are removed
-      .replace(/[-]+$/g, '') // Dashes at the end are removed
-      .replace(/-+/g, '-') // Multiple dashes in a row are deduped
+      .replace(/&/g, "and ") // & becomes and
+      .replace(/ +/g, "-") // Spaces become dashes
+      .replace(/[^\w\-]+/g, "") // Remove any non ascii characters as they might make trouble in some tools
+	  .replace(/^[-]+/g, "") // Dashes at the start are removed
+      .replace(/[-]+$/g, "") // Dashes at the end are removed
+      .replace(/-+/g, "-") // Multiple dashes in a row are deduped
```

Comment:
Some steps could be combined and simplified.

```suggestion
      .replace(/\W+/g, "-") // Replace any consecutive non ascii characters by a single dash as they might make trouble in some tools.
      .replace(/^-/g, "") // Dash at the start is removed
      .replace(/-$/g, "") // Dash at the end is removed
```

Reply:
This change will leave duplicate dashes in, won't it?

Reply:
No because it replaces all consecutive non ascii characters (`\W+`) by a single dash. `-` is included in `\W`, so in something like `hello - -- world`, `\W+` will match ` - -- ` so the result will be `hello-world`

```js
> "--  hello, this is some weird - fooo - underscore _ stuff  --".replace(/\W+/g,"-")
'-hello-this-is-some-weird-fooo-underscore-_-stuff-'
```

Reply:
Let's try and see if the tests are happy :)